src/ray/gcs/redis_client.cc

```diff
@@ -192,14 +183,8 @@ void RedisClient::Attach() {
   }
 
   instrumented_io_context &io_service = primary_context_->io_service();
-  if (options_.enable_async_conn_) {
-    asio_async_auxiliary_client_.reset(
-        new RedisAsioClient(io_service, primary_context_->async_context()));
-  }
-  if (options_.enable_subscribe_conn_) {
-    asio_subscribe_auxiliary_client_.reset(
```

Comment:
If subscribe clients are not used anymore, we should delete the declarations as well.